pkg/controller.v1/jax/jaxjob_controller.go

```diff
@@ -198,26 +197,26 @@ func (r *JAXJobReconciler) SetupWithManager(mgr ctrl.Manager, controllerThreads
 		DeleteFunc: util.OnDependentDeleteFuncGeneric(r.Expectations),
 	}
 	// inject watching for job related pod
-	if err = c.Watch(source.Kind(mgr.GetCache(), &corev1.Pod{}), eventHandler, predicates); err != nil {
+	if err = c.Watch(source.Kind[client.Object](mgr.GetCache(), &corev1.Pod{}, eventHandler, predicates)); err != nil {
```

Comment:
```suggestion
	if err = c.Watch(source.Kind[*corev1.Pod](mgr.GetCache(), &corev1.Pod{}, eventHandler, predicates)); err != nil {
```

Could we do exactly type parameter?
There are the same questions in all of Job controllers.

Reply:
I've pushed ce31d7b07c20aa5c0561df91ad76d63db1f02a85 that should cover it so we can see what that looks like.

Reply:
It looks like tests are failing, did you try to run them locally @astefanutti ?

Reply:
@astefanutti I'm happy with accepting better refactoring :)